Added support for reference-overrides.conf

corruptmemory opened the pull request:
* Added settings to suppress unnecessary warnings

patriknw commented:
LGTM, I assume that you have verified manually that it works.

To avoid future regressions it would be good with a test that that starts Play with `akka.diagnostics.checker.fail-on-warning = on` and verifies that exception is not thrown when starting the ActorSystem.
https://github.com/typesafehub/akka/blob/release-2.3-rp/akka-actor/src/main/resources/reference.conf#L753

corruptmemory commented:
Added one more commit.  The file I added for overrides contained too much (from the 2.4 version).  So  now it only contains the diagnostic overrides.

@patriknw: I did test locally to see that the configuration in `reference-overrides.conf` were loaded.  I have not checked to see if they work in the context of RP.

patriknw commented:
Play config loading is not trivial so we should test it with RP.

dotta commented:
@corruptmemory Jim, I didn't expect we would need any change in `play.api.Configuration`. Are the changes actually needed to fix the issue? (or were you being a good boy scout :-))  If the changes are needed, I think we should make them in the open source Play codebase, tag, and then pull in the changes here (the rationale being to limit as much as possible to diverge). \cc @jroper 

One more thing we would need it an integration test, because this commit is RP specific and we need to make sure it's forward backported in future releases of the PR. Is this something that you (or someone else in the RP team) could do for us? Sorry for asking, but I have unfortunately no spare cycle at the moment. If that is not possible just let me know and we will find a solution.

dotta commented on framework/src/play/src/main/scala/play/api/Configuration.scala:
```diff
@@ -37,8 +41,9 @@ object Configuration {
         devSettings.get("config.file").orElse(Option(System.getProperty("config.file")))
           .map(f => new File(f)).getOrElse(new File(appPath, "conf/application.conf"))
       }
+      val playOverridesConfig: Config = getPlayOverridesConfig(Thread.currentThread().getContextClassLoader())
```
before we were passing the classloader explicitly, so I'm unsure this change is sound. @jroper WDYT?

dotta replied:
My bad, I was looking at a wrong tag and the code in `play.api.Configuration` is quite different in Play 2.3 vs Play 2.4. So, it looks like in Play 2.3 we actually didn't have any facility to override configuration in `reference.conf`, so please disregard my above comment.

patriknw replied:
If there is no reference-overrides in Play 2.3, why do we need to suppress the warnings, there should not be any warnings with akka default settings. I must be missing something.

corruptmemory replied:
Actually you may not be. GAA! I may be confounding issues - I was looking at an issue that Johan raised related to activator but now that I think about it that may have been against 15v09.  Let me confirm, this may not be an issue.

richdougherty replied:
I can confirm these settings are present in Play 2.4, but not in Play 2.3.

dotta replied:
@patriknw You are absolutely right! We should silence the config checker only for RP releases that include Play 2.4+. Play 2.3 and earlier can't be affected by the problem.

patriknw replied:
ok, good

dotta commented:
> @corruptmemory Jim, I didn't expect we would need any change in play.api.Configuration. Are the changes actually needed to fix the issue? (or were you being a good boy scout :-)) If the changes are needed, I think we should make them in the open source Play codebase, tag, and then pull in the changes here (the rationale being to limit as much as possible to diverge). \cc @jroper

@corruptmemory I see why the changes are needed, so all good. Considering we won't have any further Play 2.3.x open source release, I also think it's fine to have the change only here. @jroper @wsargent @richdougherty Anyone disagreeing?

That said, we still need the integration test, otherwise a regression is bound to happen.

corruptmemory commented:
@dotta: mostly being a boy scout and trying to mimic what Play 2.4 was doing. I manually confirmed that the exped behavior was being observed.  Should I get time soon I will confirm that the warnings are suppressed as expeted, but my RP build has gone back to not working for me. :-(

dotta commented:
@corruptmemory Based on https://github.com/typesafehub/playframework/pull/2#discussion_r47878928,  should we close this PR? Or, am I missing something?

corruptmemory commented:
@dotta: that's correct. Closing.
